src/main/java/BotManager.java

```diff
@@ -0,0 +1,150 @@
+import java.util.ArrayList;
+
+public class BotManager {
+    private final String name;
+    private ArrayList<Task> tasks;
```

Comment:
I would consider making tasks a final attribute since you shouldn't be reassigning it later on.
```suggestion
    private final ArrayList<Task> tasks;
```